model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/PersistentUserSessionProvider.java

```diff
@@ -975,4 +982,77 @@ public static UUID createClientSessionUUID(String userSessionId, String clientId
         // This allows creating a UUID that is constant even if the entry is reloaded from the database
         return UUID.nameUUIDFromBytes((userSessionId + clientId).getBytes(StandardCharsets.UTF_8));
     }
+
+    @Override
+    public void migrate(String modelVersion) {
+        if (new ModelVersion(modelVersion).equals(new ModelVersion("25.0.0"))) {
```

Comment:
@ahus1 what does this line do? Only migrate if the running Keycloak version is 25.0.0? What would happen if a user upgrades from KC 24 to KC 25.0.1?

Reply:
This method is only called from `[MigrateTo25_0_0.java](https://github.com/keycloak/keycloak/pull/29377/files#diff-cf36311a760c7034008e1145ff898e030ba932f6df9491f4bcb5eefb897bd3f3)` which will call it when ever a Keycloak installation is upgraded from below KC 25.0.0 to that version or a later version. 

So in the example you asked for: It will be called for an upgrade from KC 24.0.x to 25.0.1

Reply:
@ahus1, the method `migrate` would be called, however, the if statement would evaluate to false with `25.0.1`, wouldn't it?

Reply:
@mhajas - the idea is to call it with a specific version from a `MigrateToXX_XX_XX.java` with a specific version parameter. This is more about "future proofing" the API, so we can call it from a future Migrate class with a different version. 

In the future, maybe each provider is called with a version, and then it can choose to do or not to do something. But we're not there, yet.

Reply:
@mhajas if I'm understanding Keycloak code properly, it invokes the migration in all the versions, 24.0.x++, 25.0.0, 25.0.1
https://github.com/keycloak/keycloak/blob/1b583a1bab2d090ac70ca9ab1528ceaeba50b3ca/model/storage-private/src/main/java/org/keycloak/storage/datastore/DefaultMigrationManager.java#L185-L197

Reply:
Ah, I missed `MigrateTo25_0_0.java` has a hardcoded version inside and not use the actual version. Sorry for confusion